Cargo.toml

```diff
@@ -12,3 +12,16 @@ members = [
     "test-buildpacks/readonly-layer-files",
     "test-buildpacks/sbom"
 ]
+
+[workspace.package]
+version = "0.10.0"
+rust-version = "1.64"
+edition = "2021"
+license = "BSD-3-Clause"
+
+[workspace.dependencies]
```

Comment:
Using `cat **/Cargo.toml | rg ' = "\d' | sort | uniq -d` I found the following additional dependencies that are shared across more than one crate:

```
cargo_metadata
fancy-regex
fastrand
serde
tempfile
```

A couple use features - which means they'd need both the workspace entry, and a more complex entry in the crate `Cargo.toml` than just `true`.

It feels like we should either include all duplicated third-party dependencies here, or else no third-party dependencies here.

In the past I've typically updated all dependencies using `cargo upgrade`, which means the versions don't get out of sync and there's no effort from having to update multiple locations (which potentially reduces one of the benefit of having inherited dependencies for these).

It also seems like over time we might adjust the dependencies of a crate and forget to re-adjust the workspace, and so end up with new duplicates, or alternatively dependencies listed in the root `Cargo.toml` that are now only used by one crate. This seems like it might end up being additional overhead / potential for inconsistency.

As such, I'm on the fence as to whether to include the third-party dependencies here - and am maybe very slightly leaning towards not including them? Though could be persuaded either way. (The libcnb* ones are a definite win however.)

Thoughts? :-)

Reply:
Yeah, honestly, I was very so-so on the third-party dependencies as well. Hard to draw a clear line when they should be in the workspace or not. The main benefit of this is that we don't have different versions of a dependency in packages. However, with dependabot, this is really a non-issue in practice.

Let remove the thrid-party dependencies for now. 👍🏻